openqaoa/optimizers/training_vqa.py

```diff
@@ -297,7 +297,7 @@ def results_dictionary(self,
         date_time = datetime.now().strftime("%d.%m.%Y_%H.%M.%S")
         file_name = f'opt_results_{date_time}' if file_name is None else file_name
         
-        self.qaoa_result = Result(self.log, self.method, self.vqa.cost_hamiltonian)
+        self.qaoa_result = Result(self.log, self.method, self.vqa.cost_hamiltonian, type(self.vqa))
```

Comment:
Maybe here you should try the `__name__` attribute? It is a bit confusing (for others) that the type of a VQA is mapped to a variable called backend. What do you think about it?

Reply:
For the asdict method of PR #144 I need it to be `type(self.vqa)`, to check its parent, so`__name__` would work.